tests/unittests/tests-spiffs/tests-spiffs.c

```diff
@@ -0,0 +1,309 @@
+    TEST_ASSERT_EQUAL_STRING(&buf[0], &r_buf[0]);
+
+    res = vfs_close(fd);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_umount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+}
+
+static void tests_spiffs_write2(void)
+{
+    char buf[1024];
+    /*char r_buf[sizeof(buf) + 2];*/
+
+    for (int i = 0; i < 1024; i++) {
+        buf[i] = 'A' + (i % 30);
+    }
+
+    int mp = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT(mp >= 0);
+
+    int res;
+    for (int j = 0; j < 10; j++) {
+        int fd = vfs_open("/test-spiffs/test.txt", O_CREAT | O_RDWR, 0);
+        TEST_ASSERT(fd >= 0);
+
+        for  (int i = 0; i < 500; i++) {
+            res = vfs_write(fd, buf, sizeof(buf));
+            TEST_ASSERT_EQUAL_INT(sizeof(buf), res);
+        }
+
+        res = vfs_lseek(fd, 0, SEEK_SET);
+        TEST_ASSERT_EQUAL_INT(0, res);
+
+        res = vfs_close(fd);
+        TEST_ASSERT_EQUAL_INT(0, res);
+
+        res = vfs_unlink("/test-spiffs/test.txt");
+        TEST_ASSERT_EQUAL_INT(0, res);
+    }
+
+    res = vfs_umount(&_test_spiffs_mount);
+    TEST_ASSERT(res >= 0);
+}
+
+static void tests_spiffs_unlink(void)
+{
+    const char buf[] = "TESTSTRING";
+
+    int res;
+    res = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    int fd = vfs_open("/test-spiffs/test.txt", O_CREAT | O_RDWR, 0);
+    TEST_ASSERT(fd >= 0);
+
+    res = vfs_write(fd, buf, sizeof(buf));
+    TEST_ASSERT(res == sizeof(buf));
+
+    res = vfs_close(fd);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_unlink("/test-spiffs/test.txt");
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_umount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+}
+
+static void tests_spiffs_readdir(void)
+{
+    const char buf0[] = "TESTSTRING";
+    const char buf1[] = "TESTTESTSTRING";
+    const char buf2[] = "TESTSTRINGSTRING";
+
+    int res;
+    res = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    int fd0 = vfs_open("/test-spiffs/test0.txt", O_CREAT | O_RDWR, 0);
+    TEST_ASSERT(fd0 >= 0);
+
+    int fd1 = vfs_open("/test-spiffs/test1.txt", O_CREAT | O_RDWR, 0);
+    TEST_ASSERT(fd1 >= 0);
+
+    int fd2 = vfs_open("/test-spiffs/a/test2.txt", O_CREAT | O_RDWR, 0);
+    TEST_ASSERT(fd2 >= 0);
+
+    res = vfs_write(fd0, buf0, sizeof(buf0));
+    TEST_ASSERT(res == sizeof(buf0));
+
+    res = vfs_write(fd1, buf1, sizeof(buf1));
+    TEST_ASSERT(res == sizeof(buf1));
+
+    res = vfs_write(fd2, buf2, sizeof(buf2));
+    TEST_ASSERT(res == sizeof(buf2));
+
+    res = vfs_close(fd0);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_close(fd1);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_close(fd2);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    vfs_DIR dirp;
+    res = vfs_opendir(&dirp, "/test-spiffs");
+
+    vfs_dirent_t entry;
+    res = vfs_readdir(&dirp, &entry);
+    TEST_ASSERT(res == 1);
+    TEST_ASSERT_EQUAL_STRING("/test0.txt", &(entry.d_name[0]));
```

Comment:
The order of these files may not be the same order as the order in which they were created. It is also possible that there are other files left over from previous runs or failed tests in the same run which interfere with this check. The test should only verify that all the created files are reported back by `readdir`, the order should be irrelevant.